File: src/Entity/Pays.php
<?php

namespace App\Entity;

use App\Repository\PaysRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;

#[ORM\Entity(repositoryClass: PaysRepository::class)]
class Pays
{

    #[ORM\Id]
    #[ORM\Column(name: 'code', type: 'string', length: 2, unique: true)]
    #[Groups(['read.Pays', 'persist.Pays', 'persist.Joueur', 'get.Joueur', 'persist.InformationsPersonelles', 'persist.InformationsPersonellesTemp', 'read.Club', 'persist.Club'])]
    private $code;

    #[ORM\Column(type: 'string', length: 255)]
    #[Groups(['read.Pays', 'persist.Pays', 'read.Club'])]
    private $nom;

    #[ORM\Column(type: 'string', length: 255, nullable: true)]
    #[Groups(['read.Pays', 'persist.Pays', 'read.Club'])]
    private $drapeau;

    #[ORM\Column(type: 'string', length: 255, nullable: true)]
    #[Groups(['read.Pays', 'persist.Pays', 'read.Club'])]
    private $nom_fr;

    public function getCode(): ?string
    {
        return $this->code;
    }

    public function setCode(string $code): self
    {
        $this->code = $code;

        return $this;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getDrapeau(): ?string
    {
        return $this->drapeau;
    }

    public function setDrapeau(?string $drapeau): self
    {
        $this->drapeau = $drapeau;

        return $this;
    }

    public function getNomFr(): ?string
    {
        return $this->nom_fr;
    }

    public function setNomFr(?string $nom_fr): self
    {
        $this->nom_fr = $nom_fr;

        return $this;
    }
}

File: public/scrapp/lib/Classes/Pays.py
class Pays:

    def __new__(cls):
        return super(Pays, cls).__new__(cls)
    
    def __init__(self):
        self.uri = None
        self.code = None
        self.nom = None
        self.drapeau = None
        self.nomFR = None

    def toJson(self, schema:str = "") -> dict:
        if schema == 'persist.Joueur':
            json = {
                "code": self.code
            }
        elif schema == 'patch.Pays.drapeau':
            json = {
                "code": self.code,
                "drapeau": self.drapeau
            }
        elif schema == 'persist.Club':
            json = self.uri
        else:
            json = {
                "code": self.code,
                "nom": self.nom,
                "drapeau": self.drapeau,
                "nomFr": self.nomFR
            }
        return json

    def fromJson(self, json: dict):
        if isinstance(json, str) :
            self.uri = json

            return self
        
        self.uri = json['@id'] if json.get('@id') else self.uri
        self.code = json['code'] if json.get('code') else self.code
        self.nom = json['nom'] if json.get('nom') else self.nom
        self.drapeau = json['drapeau'] if json.get('drapeau') else self.drapeau
        self.nomFR = json['nom_fr'] if json.get('nom_fr') else self.nomFR
        
        return self